Runtime/HistoryItemBehaviour.cs

```diff
@@ -0,0 +1,220 @@
+
+        private void OnOptionsToggleValueChanged(bool value)
+        {
+            if (value)
+            {
+                // listen if mouse button click occurs and turn off the options toggle
+                StartCoroutine(CoListenForMouseClick());
+            }
+        }
+
+        private IEnumerator CoListenForMouseClick()
+        {
+            var waitForEndOfFrame = new WaitForEndOfFrame();
+            // wait one frame to make sure we don't close the menu
+            yield return waitForEndOfFrame;
+
+            while (true)
+            {
+                if (Input.GetMouseButtonUp(0))
+                {
+                    // wait one frame to make sure to capture the click from remove button
+                    yield return waitForEndOfFrame;
+                    _optionsToggle.IsOn = false;
+                    StopAllCoroutines();
+                    yield break;
+                }
+
+                yield return null;
+            }
+        }
+
+        private async Task RequestThumbnailAsync(string id, RawImage dest, string url)
+        {
+            try
+            {
+                var texture = await ApiRequests.DownloadTextureAsync(url, cancellationToken: destroyCancellationToken);
+
+                _imageCache.TryGetValue(id, out var cachedImages);
+                var (thumbnail, depth) = cachedImages ?? new Tuple<Texture2D, Texture2D>(null, null);
+
+                if (dest == _thumbnailImage)
+                {
+                    thumbnail = texture;
+                }
+
+                if (dest == _depthImage)
+                {
+                    depth = texture;
+                }
+
+                dest.texture = texture;
+                _imageCache[id] = new Tuple<Texture2D, Texture2D>(thumbnail, depth);
+            }
+            catch (TaskCanceledException)
+            {
+                // ignored
+            }
+            catch (Exception e)
+            {
+                Debug.LogException(e);
+            }
+        }
+
+        internal async void SetItemData(ImagineResult item, Action<ImagineResult, Texture2D, Texture2D> clickCallback, Action<ImagineResult> deleteCallback, Action<ImagineResult> downloadCallback)
+        {
+            _button.interactable = false;
+
+            try
+            {
+                _imagineResult = item;
+                _clickCallback = clickCallback;
+                _deleteCallback = deleteCallback;
+                _downloadCallback = downloadCallback;
+                _descriptionText.text = $"<b>{item.skybox_style_name}</b> | {item.prompt}";
+                _timestampText.text = item.completed_at.ToString("G");
+                _likeToggle.SetIsOnWithoutNotify(item.isMyFavorite);
+                _modelBadge.SetActive(item.model == "Model 3");
+                _apiBadge.SetActive(item.api_key_id != 0);
+
+                if (!_imageCache.TryGetValue(item.obfuscated_id, out var cachedImages))
```

Comment:
maybe ok for this scenario, but to be correct you need to add a blank item to the cache before downloading to ensure you don't download twice. See style picker preview

Reply:
future: refactor image caching

Reply:
I'm used to by rest package handling caching by default. Guess I've been spoiled 😆